src/plugins/data/common/exports/escape_value.ts

```diff
@@ -21,17 +21,32 @@ type RawValue = string | object | null | undefined;
  *
  * See OWASP: https://www.owasp.org/index.php/CSV_Injection.
  */
-export function createEscapeValue(
-  quoteValues: boolean,
-  escapeFormulas: boolean
-): (val: RawValue) => string {
+export function createEscapeValue({
+  separator,
+  quoteValues,
+  escapeFormulaValues,
+}: {
+  separator: string;
+  quoteValues: boolean;
+  escapeFormulaValues: boolean;
+}): (val: RawValue) => string {
   return function escapeValue(val: RawValue) {
     if (val && typeof val === 'string') {
-      const formulasEscaped = escapeFormulas && cellHasFormulas(val) ? "'" + val : val;
-      if (quoteValues && nonAlphaNumRE.test(formulasEscaped)) {
-        return `"${formulasEscaped.replace(allDoubleQuoteRE, '""')}"`;
+      const formulasEscaped = escapeFormulaValues && cellHasFormulas(val) ? "'" + val : val;
```

Comment:
This PR fixes a problem of escape for the `,` separator. But here we allow _any_ string to be passed as separator, even the empty string `""`. 
I am not sure we want to wrap in quotes if the separator is empty or the letter `a`, that is quite greedy 😊 

Can we have a list of chars that would require wrapping? Something like

```ts
const doWrapWithQuotes = [',', ';'].includes(separator);
// Only wrap if flag is true
```

Reply:
good catch, an other possibility is to force it via typescript with something like:
```ts
function createEscapeValue({
  separator,
  quoteValues,
  escapeFormulaValues,
}: {
  separator: ',' | ';';
  quoteValues: boolean;
  escapeFormulaValues: boolean;
})
```

Reply:
From a quick search the common delimiters for CSV and similar formats are:
```
const delimiters = [',', ';', '\t',' ', '|'];
```

I can change the PR to wrap the value only if these delimiters are used. Would it work for you @sebelga ?

Reply:
I've checked where the separator value comes from. It arrives from the Settings `csv:separator` and there is no check for that value, so the typescript  only will not work here.
I think checking for the most common separators is safe and doesn't break the common sense configurations

Reply:
Yeah I think we don't want to rely only on TS for this kind of validations that modifies values. 

> I can change the PR to wrap the value only if these delimiters are used. 

That'd be great 👍

Reply:
I also guess it is the safest bet 👍